File: knowledge/scaffolds/web-backend/spring-boot-java/src/main/java/com/example/myapp/controller/HelloController.java
package com.example.myapp.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/api")
public class HelloController {

    @GetMapping("/hello")
    public ResponseEntity<Map<String, String>> hello() {
        return ResponseEntity.ok(Map.of("message", "Hello, World!"));
    }

    @GetMapping("/hello/{name}")
    public ResponseEntity<Map<String, String>> helloName(@PathVariable String name) {
        return ResponseEntity.ok(Map.of("message", "Hello, " + name + "!"));
    }

}

File: knowledge/scaffolds/web-backend/spring-boot-kotlin/src/main/kotlin/com/example/myapp/controller/HelloController.kt
package com.example.myapp.controller

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class HelloController {

    @GetMapping("/hello")
    fun hello(): ResponseEntity<Map<String, String>> {
        return ResponseEntity.ok(mapOf("message" to "Hello, World!"))
    }

    @GetMapping("/hello/{name}")
    fun helloName(@PathVariable name: String): ResponseEntity<Map<String, String>> {
        return ResponseEntity.ok(mapOf("message" to "Hello, $name!"))
    }

}
